### DISCUSSION TITLE
Only enable linters activated in project-specific files

### DISCUSSION THREAD

**Discussion by sigvaldm** [w0rp/ale#4466] [Q&A - Ask for help with problems] [answered]
Dear community, I like ALE, but I don't like the overwhelming amount of information from linters not currently in use in projects I work on.

Currently I specify a list of linters using `let g:ale_linters = ...`, but, when I switch to another project where one of the linters is not in use, I get a ton of warnings. I primarily work on Python projects, and the linters in use typically have files in the root folder of the project. Flake8 have `.flake8`, Pylint have `.pylintrc`, and many linters can be configured in `pyproject.toml`. Is it possible to disable all linters by default, and only activate linters for files which are covered by one such config file?

**Comment by hsanson** [comment, marked as answer]
Maybe `g:ale_linters_explicit` does what you want.  It will disable everything by default and only run linters defined in g:ale_linters.

I do not think ALE will ever depend on linters configuration files (e.g. .flake8) to determine which linters to run or not. There are too many possibilities and would be rather difficult to get something that works for everyone.

**Reply by w0rp** [reply]
That's the setting to use.

**Reply by sigvaldm** [reply]
Not depending on linter configuration files is an understandable design choice. `g:ale_linters_explicit` is helpful indeed. I'll add that I found the plugin [embear/vim-localvimrc](https://github.com/embear/vim-localvimrc) useful for project-specific settings (so long as you're allowed to store settings like these in project repositories). Having it installed, I add the following lines to `.vimrc` (not necessarily next to one another, but at logical places):

```
let g:ale_linters_explicit = 1
let g:localvimrc_persistent = 1
```

localvimrc allows project-specific config file `.lvimrc` that will apply for all files in that directory and subfolders. It is similar to the Vim `exrc` functionality, but it asks before loading local config, such as to prevent potential security issues. With the persistent setting above, it will only ask once per config file. As an example, I can set the linters for a project inside the `.lvimrc` file like so:

```
let g:ale_linters = {'python': ['flake8', 'mypy']}
```
